src/reanimated2/animation/spring.ts

```diff
@@ -49,28 +51,17 @@ export const withSpring = ((
       reduceMotion: undefined,
     } as const;
 
-    const config: Record<keyof SpringConfig, any> & SpringConfigInner = {
+    const config: DefaultSpringConfig & SpringConfigInner = {
       ...defaultConfig,
       ...userConfig,
       useDuration: !!(userConfig?.duration || userConfig?.dampingRatio),
-      configIsInvalid: false,
+      skipAnimation: false,
     };
 
-    if (
-      [
-        config.stiffness,
-        config.damping,
-        config.duration,
-        config.dampingRatio,
-        config.restDisplacementThreshold,
-        config.restSpeedThreshold,
-      ].some((x) => x <= 0) ||
-      config.mass === 0
-    ) {
-      config.configIsInvalid = true;
-      console.warn(
-        "You have provided invalid spring animation configuration! \n Value of stiffness, damping, duration and damping ratio must be greater than zero, and mass can't equal zero."
-      );
+    config.skipAnimation = validateConfig(config);
+
+    if (config.duration === 0) {
+      config.skipAnimation = true;
```

Comment:
If the current value is 100, the target value is 200 and duration is 0, do we set the final value in such case?

Reply:
Yes it is.
In this line we set `skipAnimation` flag, which is used in this clause:
```
      if (config.skipAnimation) {
        // If we use duration we want to wait the provided time before stopping
        if (config.useDuration) return false;
        else {
          animation.current = toValue;
          animation.lastTimestamp = 0;
          return true;
        }
      }
```
Therefore the animation without duration sets the value inside this if condition and animation with `duration = 0` sets the final value in next iteration, here:

```javascript

      if (config.useDuration && timeFromStart >= config.duration) {
        animation.current = toValue;
        // clear lastTimestamp to avoid using stale value by the next spring animation that starts after this one
        animation.lastTimestamp = 0;
        return true;
      }
```

Reply:
@tomekzaw  Now I regret the decision to wait the duration time for invalid config and I think we should always update the value immediately, WDYT?

Reply:
Sounds ok

Reply:
@tomekzaw Is this a change for this PR or for a separate one?